From c58cbd2cdcdf3484442f65b7c9e072d85424d31d Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Sat, 12 Jul 2014 17:07:32 +0200
Subject: [PATCH] Make sure generated service files get a .service suffix

ls2 expects all files to end with a proper .service suffix so we have to warant that when
generating them on the fly.

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 serviceinstall.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/serviceinstall.cpp b/serviceinstall.cpp
index efe53b7..f1080ac 100644
--- a/serviceinstall.cpp
+++ b/serviceinstall.cpp
@@ -77,7 +77,7 @@ static string getPrivateEndpointDirectory(string root)
 static void generateEndpoint(string id, string serviceDirectory, string endpointDirectory)
 {
 	string line;
-	string destinationPath = endpointDirectory + "/" + id;
+	string destinationPath = endpointDirectory + "/" + id + ".service";
 	DIR *d;
 
 	if ((d=opendir(endpointDirectory.c_str())) == NULL) {
@@ -106,7 +106,7 @@ static void generateEndpoint(string id, string serviceDirectory, string endpoint
 
 static void deleteEndpoint(string id, string endpointDirectory)
 {
-	string destinationPath = endpointDirectory + "/" + id;
+	string destinationPath = endpointDirectory + "/" + id + ".service";
 	if (fexists(destinationPath)) {
 		DBG("Removing %s", destinationPath.c_str());
 		unlink(destinationPath.c_str());
-- 
2.37.3.windows.1

